### ISSUE TITLE
[java] UnusedPrivateMethod false positive with method references

### ISSUE THREAD

**Issue by sd-ebrukanat** [pmd/pmd#5455]
**Affects PMD Version:** 
7.8.0 

**Rule:**
Rule: [UnusedPrivateMethod](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**
private method referenced by method reference operators not getting detected as used

**Code Sample demonstrating the issue:**

```java
 import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;
import reactor.core.publisher.Mono;
import reactor.function.TupleUtils;
import java.util.Optional;
import static org.assertj.core.api.Assertions.assertThat;
@ExtendWith(MockitoExtension.class)
class PmdTest2 {
    @Test
    void test() {
        var res = Mono.zip(Mono.just(Optional.of("Some test")), Mono.just(TestClass.of("some 2")))
                .flatMap(TupleUtils.function((par1, par2) -> Mono.just(createProperty(par1, par2))))
                .map(objectProperty -> addRequired(objectProperty))
                .block();
        assertThat(res.required).isEqualTo("pop");
        assertThat(res.val).isEqualTo("Some test");
    }
    private TestClass createProperty(Optional<String> par1, TestClass objectProperty) {
        return TestClass.of(par1.orElse("par5"));
    }
    private TestClass addRequired(TestClass objectProperty) {
        return objectProperty.withRequiredProperty("pop");
    }
    private static class TestClass {
        private String required;
        private String val;
        TestClass(String val) {
            this.val = val;
        }
        public static TestClass of(String val) {
            return new TestClass(val);
        }
        public TestClass withRequiredProperty(String s) {
            required = s;
            return this;
        }
        public String getRequired() {
            return required;
        }
        public String getVal() {
            return val;
        }
    }
}
 
```

**Expected outcome:**
It is not an unused private method

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by punkratz312** [comment]
Works on my machine: `0 problem(s) found.`

I also had an issue where live coding produced false positives. The code was copied 1-to-1 into a test, and surprisingly, it was correctly interpreted there.

<img width="367" alt="Image" src="https://github.com/user-attachments/assets/16259337-5d58-466c-9d25-c50f44869b2c" />

`UnusedPrivateMethod.xml`
```
    <test-code>
        <description>PmdTest2</description>
<!--        <expected-problems>0</expected-problems>-->
        <expected-problems>1</expected-problems>
        <code><![CDATA[
 import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;
import reactor.core.publisher.Mono;
import reactor.function.TupleUtils;
import java.util.Optional;
import static org.assertj.core.api.Assertions.assertThat;
@ExtendWith(MockitoExtension.class)
class PmdTest2 {
    @Test
    void test() {
        var res = Mono.zip(Mono.just(Optional.of("Some test")), Mono.just(TestClass.of("some 2")))
                .flatMap(TupleUtils.function((par1, par2) -> Mono.just(createProperty(par1, par2))))
                .map(objectProperty -> addRequired(objectProperty))
                .block();
        assertThat(res.required).isEqualTo("pop");
        assertThat(res.val).isEqualTo("Some test");
    }
    private TestClass createProperty(Optional<String> par1, TestClass objectProperty) {
        return TestClass.of(par1.orElse("par5"));
    }
    private TestClass addRequired(TestClass objectProperty) {
        return objectProperty.withRequiredProperty("pop");
    }
    private static class TestClass {
        private String required;
        private String val;
        TestClass(String val) {
            this.val = val;
        }
        public static TestClass of(String val) {
            return new TestClass(val);
        }
        public TestClass withRequiredProperty(String s) {
            required = s;
            return this;
        }
        public String getRequired() {
            return required;
        }
        public String getVal() {
            return val;
        }
    }
}

        ]]></code>
    </test-code>
```

**Comment by punkratz312** [comment]
https://github.com/pmd/pmd/pull/5469/files

**Comment by adangel** [comment]
Note, the description talks about method references ("private method referenced by method reference operators not getting detected as used"), but there are no method references in the sample code. So probably, that's the wrong code sample.

**Comment by adangel** [comment]
Closing this as it is not reproducible. Feel free to reopen if you have a code sample the reproduces the problem.
